src/utils/index.js

```diff
@@ -11,28 +9,58 @@ export const API_URL = process.env.REACT_APP_STAGE  === 'production'
 
 export const AWS_CONFIG = aws_config
 
-export const API_KEY = process.env.REACT_APP_STAGE  === 'production'
+const API_URL = process.env.REACT_APP_STAGE  === 'production'
+    ? process.env.REACT_APP_PROD_API
+    : process.env.REACT_APP_STAGING_API
+    
+const API_KEY = process.env.REACT_APP_STAGE  === 'production'
     ? process.env.REACT_APP_PROD_API_KEY
     : process.env.REACT_APP_STAGING_API_KEY
 
-export function fetchBackend(endpoint, method, body) {
+export function fetchBackend(endpoint, method, data) {
     let headers;
     if (method === 'POST') {
         headers = {
             'x-api-key': API_KEY,
             Accept: 'application/json',
             'Content-Type': 'application/json',
         }
-    } else { // method === 'GET'
+    } else {
         headers = {
             'x-api-key': API_KEY
         }
     }
+    const body = JSON.stringify(data)
+    let status;
     return fetch(API_URL + endpoint, {method, headers, body})
+        .then(response => {
+            status = response.status;
+            return response.json()
+        })
+        .then(response => {
+            // Actually throw an error (so catch block will run) when the response is an error
+            if (status < 200 || status >= 300) {
+                return Promise.reject({
```

Comment:
Should this be Promise.reject or should it just be an error thrown? If we do Promise.reject maybe we should also use Promise.resolve

Reply:
not sure when to reject or throw an error. I picked reject because otherwise it would return a Promise.resolve (which is also technically the return type of the success case)

Reply:
oh I see so it already defaults to promise.resolve?

Reply:
Thats what I thought but I don't know exactly how to check. `.then` is evaluated correctly from this helper though.

Reply:
Adding .resolve() seems to also work